src/fenwick.jl

```diff
@@ -0,0 +1,96 @@
+import Base.getindex
+"""
+    FenwickTree{T}(n)
+    
+Constructs a [`FenwickTree`](https://en.wikipedia.org/wiki/Fenwick_tree) of length `n`.
+ 
+"""
+mutable struct FenwickTree{T}
+    bit::Vector{T}
+    n::Integer
+end
+
+FenwickTree{T}() where T = FenwickTree{T}(0)
+FenwickTree{T}(n::Integer) where T = FenwickTree{T}(zeros(T, n), n)
+
+"""
+    FenwickTree(arr::AbstractArray) 
+    
+Constructs a [`FenwickTree`](https://en.wikipedia.org/wiki/Fenwick_tree) from `arr`.
+ 
+"""
+function FenwickTree(a::T) where T <: AbstractVector
+    U = eltype(a)
+    n = size(a)[1]
+    bit = FenwickTree{U}(n)
+    @boundscheck for i = 1:n
```

Comment:
This should still be `@inbounds`. `@boundscheck` is used for error checking, e.g.
```julia
@boundscheck condition || error
@inbounds for ...
```